nrich-search/README.md

```diff
@@ -175,41 +201,167 @@ public class CarSearchService {
-used for query.
-
-`subqueryConfigurationList` is useful when we want to search entities without direct association i.e. if we would like to search User based on a Role and User has no direct association to role instead
-we mapped that connection by using UserRole entity.
+### `NaturalIdSearchExecturor`
 
-```
+`NaturalIdSearchExecturor` only works with Hibernate JPA implementation and enables users to search entities by properties annotated with `@NaturalId` annotation.
+It is useful when entity has, beside its primary key, a natural key (akin to `registrationNumber` in entity `Car`).
+
+---
+
+> Users are able to use all types of search executors together on one JPA repository. This is achieved by extending `SearchExecutor`, `StringSearchExecutor` and `NaturalIdSearchExecturor` in
+> user defined JPA repository. Example of such a repository would be:
+>
+>```java
+>    public interface CarSearchDemoRepository extends JpaRepository<Car, Long>, SearchExecutor<Car>, StringSearchExecutor<Car>, NaturalIdSearchExecutor<Car> {
+>
+>    }
+>```
+
+
+## `SearchConfiguration`
+
+`SearchConfiguration` has a lot of options that let users customize query creation and result resolution.
+
+##### Defining result class
+
+If a regular class needs to be returned instead of projection,`.resultClass(CarSearchResult.class)` can be omitted and return class will be resolved from repository type parameter.
+Instead of a result class users can also manually define a list of projections using `projectionList` property.
+
+#### Defining table joins
+
+Users can define joins (fetches) (so no data is fetched in one select) by using `joinList`.
+Supported joins are **inner joins** and **left outer joins**.
+Joins are supplied by using the class `SearchJoin`.
 
+#### Defining property mapping
+
+`resolvePropertyMappingUsingPrefix` resolves properties by prefix, but it is also possible to write explicit property mapping from search request to searched entity by using`propertyMappingList`.
+
+#### Default operators
+
+Default operators (see `DefaultSearchOperator`) are for String _**ILIKE**_ (uses `criteriaBuilder.like` with lower call before), for range search _**GT**_ (
+uses `criteriaBuilder.greaterThan`), _**LT**_ (uses `criteriaBuilder.lessThan`), _**GE**_ (uses `criteriaBuilder.greaterThanOrEqualTo`), _**LE**_ (uses `criteriaBuilder.lessThanOrEqualTo`)
+and for all other classes _**EQ**_ but this can be overridden either on type level or on property level by using `searchOperatorOverrideList`.
+
+Queries on plural associations are done using exists query (to avoid duplicate results) but that can be overridden by property `pluralAssociationRestrictionType`.
+Additional restrictions (not dependent on data in search request, for example security restrictions) can be specified using `additionalRestrictionResolverList`.
+
+#### Setting **and** operator or **or** operator
```

Comment:
Maybe change to `Setting and/or operator`. This `or or` is confusing.

Reply:
Done, its much clearer.